src/shared/views/Mapper/CardBody.js

```diff
@@ -419,15 +398,34 @@ const mapStateToProps = (state, ownProps) => ({
 export default compose(
     firebaseConnect((props) => {
         if (props.group) {
-            const { groupId } = props.group;
+            const { groupId, projectId } = props.group;
             const prefix = props.tutorial ? 'tutorial' : 'projects';
-            return [
-                {
-                    type: 'once',
-                    path: `v2/tasks/${props.projectId}/${groupId}`,
-                    storeAs: `${prefix}/${props.projectId}/groups/${groupId}/tasks`,
-                },
-            ];
+            // we only load tasks for tutorial mode, otherwise we can just interpolate them
+            // from the group info
+            if (groupId !== undefined && props.tutorial) {
+                const r = props.results;
+                // also wait for the startTime timestamp to be set (by START_GROUP)
```

Comment:
I like that you put so much comments in this file. This helped a lot. :)

Reply:
yes, this is how it should be in every file.